lms/static/scripts/frontend_apps/config.js

```diff
@@ -82,31 +90,37 @@ import { createContext } from 'preact';
  */
 
 /**
- * @typedef {'blackboard_missing_integration' |
- *  'canvas_invalid_scope' } OAuthErrorCode
- */
-
-/**
- * Configuration for the error dialog shown if authorizing access to an
- * OAuth API fails.
- *
- * @typedef OAuthErrorConfig
- * @prop {string|null} authUrl
- * @prop {string} errorDetails
- * @prop {OAuthErrorCode|null} errorCode
- * @prop {string[]} canvasScopes
+ *
+ * @typedef {'blackboard_missing_integration' | 'canvas_invalid_scope' | 'reused_consumer_key'} ConfigErrorCode
+ *
+ * Base interface for error information provided through this configuration object.
+ *
+ * @typedef ConfigErrorBase
+ * @prop {ConfigErrorCode} [errorCode]
```

Comment:
Are `errorCode` and `errorDetails` really optional?

Something I noticed in testing this is that TS is able to better error checking for non optional fields. The code below gives an error if `errorCode` is non-optional but nothing if it is optional:

```js
if (config.errorCode === 'not-a-valid-error-code') {
...
}
```

I wonder if there is a config setting that gives us more control over this.